### PR TITLE
ED-3071 get company by Companies House ID

### PR DESCRIPTION
**Pull request by kowalcj0**
This ticket: [ED-3071](https://uktrade.atlassian.net/browse/ED-3071)

### CODE REVIEW

File: testapi/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from django.core.urlresolvers import reverse
+from rest_framework import status
+
+from company.tests import factories
+
+
+@pytest.mark.django_db
+def test_get_existing_company_by_ch_id(authed_client, authed_supplier):
+    url = reverse(
+        'company_by_ch_id', kwargs={"ch_id": authed_supplier.company.number})
```
Comment (sdonk) [inline testapi/tests/test_views.py]: Please don't mix `'` and `"`

File: testapi/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        name='Test Company', date_of_creation=datetime.date(2000, 10, 10),
+        email_address='test@user.com', verification_code=verification_code,
+        is_verification_letter_sent=False
+    )
+    authed_supplier.company = company
+    authed_supplier.save()
+    company.refresh_from_db()
+    assert company.verification_code
+    url = reverse(
+        'company_by_ch_id', kwargs={"ch_id": authed_supplier.company.number})
```
Comment (sdonk) [inline testapi/tests/test_views.py]: same here

File: README.md [hunk 3]
```diff
@@ -110,3 +110,84 @@ Therefore to make cookie sharing work in development we need the apps to be runn
...
+cmd="create_registered_company ch_id_max_8chrs" make debug_manage
+```
+
+## Linux setup
+
+Tested on Fedora 27
+
+### Install postgres
+```shell
+sudo dnf install \
```
Comment (richtier) [inline README.md]: `dnf` seems to be fedora-specific. The project is used by devs using Ubuntu and Mac, so we cant be platform-restrictive.

we could include instructions here for running a docker container running postgres? (`docker-compose up postgres` from within directory-api)

File: README.md [hunk 4]
```diff
@@ -110,3 +110,84 @@ Therefore to make cookie sharing work in development we need the apps to be runn
...
+```
+
+Then start the `postgres`:
+```shell
+sudo systemctl start postgresql
+```
+
+### Run elasticsearch
+
+Due to an elasticsearch [bug](https://github.com/elastic/elasticsearch/issues/23486) in handling `cgroups` version 5.1.2 won't work on Fedora.
```
Comment (richtier) [inline README.md]: which is a reallly strong argument for using docker. the command is `docker-compose up elasticsearch` (from within directory-api)

File: api/settings.py [hunk 5]
```diff
@@ -508,3 +509,7 @@
 CSV_DUMP_AUTH_TOKEN = os.environ['CSV_DUMP_AUTH_TOKEN']
 BUYERS_CSV_FILE_NAME = 'find-a-buyer-buyers.csv'
 SUPPLIERS_CSV_FILE_NAME = 'find-a-buyer-suppliers.csv'
+
+# testing api
+FEATURE_TEST_API_ENABLE = os.getenv(
```
Comment (richtier) [inline api/settings.py]: for consistency this should be `FEATURE_TEST_API_ENABLED` (with a D at the end) :)

File: testapi/views.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    permission_classes = [SignatureCheckPermission]
+    lookup_field = 'number'
+    http_method_names = ('get', )
+
+    def dispatch(self, *args, **kwargs):
+        if not settings.FEATURE_TEST_API_ENABLE:
+            raise Http404
+        return super().dispatch(*args, **kwargs)
+
+    def get(self, request: Request, ch_id: str, format: str = None):
```
Comment (richtier) [inline testapi/views.py]: we don't use the type hinting anywhere else in API. If we're adding it, please create a refactoring ticket for adding it everywhere in API. https://uktrade.atlassian.net/browse/ED-3024

File: api/settings.py (hunk 5 above)
Comment (kowalcj0) [reply api/settings.py]: indeed

File: README.md (hunk 3 above)
Comment (kowalcj0) [reply README.md]: I understand that such instructions can get pretty noisy, but why wasting time on searching information relevant to this project on other sites?
The first time I had to set up an env for this project, I literally wasted few hours on finding information how to get things done. Not sure if it's worth for someone else to go through the same or similar process again.
@richtier I can put Linux specific installation instructions in a separate file and simply link it here.

File: README.md (hunk 4 above)
Comment (kowalcj0) [reply README.md]: @richtier This paragraph refers to a problem with running `elasticsearch` with docker.
Unfortunately containers are not 100% independent from the host OS.
I haven't tried running it outside the container.

File: testapi/views.py (hunk 6 above)
Comment (kowalcj0) [reply testapi/views.py]: @richtier Was "type hinting" ever discussed among us?
I like the idea of it. Moreover, some IDEs like `PyCharm`, make use of it, and highlight potential type casting issues before you even run the application.

Comment (kowalcj0) [reply testapi/views.py]: Official `Python Type Checking Guide Documentation, Release 1.0`
https://media.readthedocs.org/pdf/python-type-checking/latest/python-type-checking.pdf

File: testapi/tests/test_views.py (hunk 1 above)
Comment (kowalcj0) [reply testapi/tests/test_views.py]: corrected

File: testapi/tests/test_views.py (hunk 2 above)
Comment (kowalcj0) [reply testapi/tests/test_views.py]: corrected

File: README.md (hunk 3 above)
Comment (kowalcj0) [reply README.md]: @richtier I moved those instructions to a separate `docs/LINUX.md` file

File: api/settings.py (hunk 5 above)
Comment (kowalcj0) [reply api/settings.py]: fixed

**Pull request merged**